lib/sidebar.rb

```diff
@@ -29,178 +30,111 @@ def pre_render(page)
-        next if value[:title].nil?
-        next if value[:hide_from_sidebar]
-
-        subgroups = merged.select { |subgroup_key, _subgroup_value| subgroup_key.include? key and subgroup_key != key and key != '/' }
-
-        active = active?(filename, key)
-        # puts "#{filename}, #{key}" if active
-        item_class = active ? 'nav-group active' : 'nav-group'
+    def write_sidebar_to_files(files)
+      files.each do |file|
+        sanitized_filename = file[:sanitized_filename]
+        filename = file[:filename]
+        doc = file[:doc]
+        sidebar_markup = render(sanitized_filename)
 
-        child = "<li class=\"#{item_class}\">"
-        child << "<div class=\"nav-group-heading\"><i class=\"material-icons\">arrow_right</i><span>#{value[:title].split('–').first}</span></div>"
+        next if sidebar_markup.nil?
 
-        child << '<ul class="nav-ul">'
+        sidebar_containers = find_containers(doc, filename)
 
-        subgroup = generateSubgroup(filename, key, value, subgroups, 2)
+        next if sidebar_containers.nil?
 
-        child << subgroup
-
-        if subgroups.any?
-          subgroups.select { |subgroup_key, _subgroup_value| subgroup_key.split('/').length <= 3 }.each do |subgroup_key, subgroup_value|
-            subgroup = generateSubgroup(filename, subgroup_key, subgroup_value, subgroups, 2)
-            child << subgroup
-          end
+        sidebar_containers.each do |sidebar_container|
+          sidebar_container.inner_html = sidebar_markup
         end
 
-        child << '</ul>'
-        child << '</li>'
-        sidebar << child
+        Jekyll.logger.debug("   Writing Sidebar: #{filename}")
+        File.open(filename, 'w') { |f| f.write(doc.to_html(encoding: 'UTF-8')) }
       end
-
-      File.open('_site/sidebar.html', 'w') { |f| f.write(sidebar) }
-      sidebar
     end
 
-    def active?(filename, url, exact = false)
-      if filename == '/' || url == '/'
-        if filename == '/' && url == '/'
-          return true
-        else
-          return false
-        end
-      end
-
-      exact ? filename == url : filename.start_with?(url)
-    end
+    def render(filename)
+      sidebar_markup = nil
 
-    def merge(hash1, hash2)
-      all_keys = hash1.keys | hash2.keys
-      result_hash = {}
+      begin
+        pages = @pages_pre_render \
+                .safe_merge(@pages_with_headers) \
+                .sort_by { |_, page| page[:menu_order] }
 
-      all_keys.each do |key|
-        hash_value = {}
+        builder = SidebarBuilder.new(filename, pages)
 
-        if hash1.key? key
-          value = hash1[key]
+        sidebar_markup = builder.build
 
-          hash_value = value unless value.nil?
-        end
+        File.open('_site/sidebar.html', 'w') { |f| f.write(sidebar_markup) }
+      rescue StandardError => e
+        Jekyll.logger.error("           Sidebar: Unable to render sidebar for '#{filename}'.")
```

Comment:
Codacy found an issue: [Line is too long. [94/80]](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-design-guide-jekyll-theme/pullRequest?prid=6198407)